tests/tests/metrics.rs

```diff
@@ -88,14 +86,24 @@ where
 			{
 				let mut n = counter.lock().unwrap();
 				n.requests.1 += 1;
-				if rp.is_success() {
+				if rp.as_ref().is_ok_and(|r| r.is_success()) {
 					n.calls.get_mut(&name).unwrap().1.push(id.into_owned());
 				}
 			}
 
 			rp
 		}
-		.boxed()
+	}
+
+	fn batch<'a>(&self, batch: Batch<'a>) -> impl Future<Output = Result<Self::Response, Self::Error>> + Send + 'a {
+		self.service.batch(batch)
+	}
+
+	fn notification<'a>(
+		&self,
+		n: Notification<'a>,
+	) -> impl Future<Output = Result<Self::Response, Self::Error>> + Send + 'a {
+		self.service.notification(n)
 	}
```

Comment:
If these aren't necessary, maybe better to `panic!("not used in tests")` or something, so it's super clear in case we ever add a test which makes a batch call or whatever and get confused why nothing changes?